### PR TITLE
Stream basics - Messages (remade)

### PR DESCRIPTION
**Pull request by jamesm-macrometa**
This PR revises the Streams topic "Messages" to update it to our current style, and also revise for polish and formatting.

Remade from https://github.com/Macrometacorp/docs/pull/633

### CODE REVIEW

File: docs/cep/index.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ The stream processing engine receives data event-by-event and processes them in
 - Process them to transform, enrich, and generate insights.
 - Publish them to multiple types of sinks.
 
-To use stream processor, you need to write the processing logic as a stream application using streaming SQL language which is discussed in the [Stream Query Guide](query-guide/index.md).
+To use stream processor, you need to write the processing logic as a stream application using streaming SQL language which is discussed in the [Stream Query Guide](query-guide/).
```
Comment (diana-macrometa) [inline docs/cep/index.md]:
```suggestion
To use process streams, you need to write the processing logic as a stream worker query, which is discussed in [Stream Worker Queries](query-guide/).
```

File: docs/cep/index.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ The stream processing engine receives data event-by-event and processes them in
...
 
 When the stream application is published, it:
```
Comment (diana-macrometa) [inline docs/cep/index.md]:
```suggestion
When the stream worker is published, it:
```

File: docs/streams/stream-basics/message-queues/index.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+sidebar_position: 1
+title: Message Queues
+---
+
+A message queue automatically balances loads across all consumers for a given stream. Best practice is to use multiple consumers.
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/message-queues/index.md:6]: How do I create multiple consumers? Or even one?

File: docs/streams/stream-basics/message-queues/index.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+Each stream has a receiver queue that determines how many messages the consumer will attempt to fetch at a time. A receiver queue of 1,000 (the default), for example, means that the consumer will attempt to process 1,000 messages from the stream's backlog upon connection. Setting the receiver queue to zero essentially means ensuring that each consumer is only doing one thing at a time.
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/message-queues/index.md]: How do I set the queue to zero?

File: docs/streams/stream-basics/message-queues/index.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+The downside to restricting the receiver queue size of consumers is that that limits the potential throughput of those consumers. Whether the performance/control trade-off is worthwhile will depend on your use case.
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/message-queues/index.md]: Try to reword this sentence to get rid of "that that."

File: docs/streams/stream-basics/message-queues/set-message-queue.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
+---
+sidebar_position: 10
+title: Set Message Queue
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+To use a stream as a message queue:
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/message-queues/set-message-queue.md:11]:
```suggestion
This page shows you several ways to use a stream as a message queue.

To use a stream as a message queue:
```

File: docs/streams/stream-basics/messages.md [hunk 7]
```diff
@@ -3,88 +3,34 @@ sidebar_position: 20
...
-
-```js
-stream.consumer("my-subscription", "test.macrometa.io", {
-  subscriptionType: Shared,
-  receiverQueueSize: 1000,
-}
-```
-
-## Message Retention and Expiry
+- **Value -** The data carried by the message. All GDN stream messages carry raw bytes, although message data can also conform to data schemas in the future.
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/messages.md]: What does "conform to data schemas in the future" mean?

File: docs/streams/stream-basics/messages.md [hunk 8]
```diff
@@ -3,88 +3,34 @@ sidebar_position: 20
...
 GDN streams has two features, however, that enable you to override this default behavior:
 
 - Message _retention_ allows you to store messages that have been acknowledged by a consumer.
-- Message _expiry_ allows you to set a time to live (TTL) for messages that have not yet been acknowledged.
+- Message _expiration_ allows you to set a time to live (TTL) for messages that have not yet been acknowledged.
 
-:::note
-All message retention and expiry is managed at the geofabric level.
-:::
+All message retention and expiration is managed at the geofabric level. You can also add a [message queue](./message-queues/index.md) to a specific stream limit the number of messages in the backlog.
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/messages.md:30]: index.md

**Review by diana-macrometa — APPROVED** [review]
There are a couple questions that will need to get answered before this project is done, but they do not need to be answered in this PR.

**Comment by jamesm-macrometa** [comment]
@diana-macrometa My bad, this was supposed to be in draft status still! I appreciate the feedback 😉

File: docs/streams/stream-basics/message-queues/index.md (hunk 3 above)
Comment (jamesm-macrometa) [reply docs/streams/stream-basics/message-queues/index.md:6]: @diana-macrometa I'll update this to add a link once we've merged the other PR which adds the "Create Consumers" topic

**Comment by jamesm-macrometa** [comment]
@diana-macrometa Significantly revised after your last review, please take another look when you have bandwidth.  Thanks!

File: docs/cep/index.md [hunk 9]
```diff
@@ -32,9 +32,9 @@ You can process streams to perform the following actions with your data:
...
 - Integrating stream data and trigger actions based on the data. This can be a single service request or a complex enterprise integration flow.
 - Consume and publish events.
-- Run pre-made and custom [functions](query-guide/functions/index.md).
+- Run pre-made and custom [functions](query-guide/functions/).
 - Write custom [JavaScript functions](query-guide/custom-script-functions.md) to interact with your streams.
 - Query, modify, and join the data stored in [tables](table/index.md) which support primary key constraints and indexing.
 - Rule processing based on single event using [`filter`](query-guide/query.md#filter) operator, `if-then-else` and `match` [functions](query-guide/query.md#function), and many others.
 
-These actions allow you to build robust global data processing and integration pipelines at the edge by combining powerful stream processing, multi-model database and geo-replicated streams capabilities.
+These actions allow you to build robust global data processing and integration pipelines at the edge by combining powerful stream processing, multi-model database and geo-replicated streams capabilities.
```
Comment (diana-macrometa) [inline docs/cep/index.md:40]: At some point, we should get some guidance from marketing to make sure this content is in line with our current official verbiage for the GDN.

File: docs/streams/stream-basics/message-queues/index.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+---
+sidebar_position: 1
+title: Message Queues
+---
+
+A message queue automatically balances loads across all consumers for a given stream. Best practice is to use multiple consumers.
+
+Each stream has a receiver queue that determines how many messages the consumer will attempt to fetch at a time. A consumer with a receiver queue of `1000` (default) attempts to process 1,000 messages from the stream's backlog upon connection. Setting the receiver queue to zero verifies that each consumer only processes one message at a time. Refer to the [Set Message Queue](set-message-queue.md) topic for more details.
+
+When you restrict the receiver queue size, you limit the potential throughput of those consumers. You should decide if you prefer greater performance or greater control.
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/message-queues/index.md]: Not sure about "should" in this sentence, seems kind of wishy-washy. As a reader, I'm not sure how I would interpret that.

File: docs/streams/stream-basics/message-queues/set-message-queue.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+1. Establish a [shared subscription](../subscriptions#shared) and use the same subscription name as the other consumers. 
+1. Set the **receiver queue** for the consumers.
+
+When you subscribe to a stream, use the following configuration settings depending on your SDK:
+
+<Tabs groupId="operating-systems">
+<TabItem value="py" label="Python SDK">
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/message-queues/set-message-queue.md]: Make sure the order of these tabs is consistent throughout. I am pretty sure that these are reversed from the other PR I just edited.

File: docs/streams/stream-basics/messages.md [hunk 12]
```diff
@@ -3,88 +3,34 @@ sidebar_position: 20
 title: Messages
 ---
 
-Messages are the basic unit of GDN streams. Producers publish messages to streams, and consumers then consume them streams and acknowledge when the message has been processed.
+Messages are the basic unit of GDN streams. Producers publish messages to streams, and consumers process the message and send and acknowledgment.
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/messages.md]:
```suggestion
Messages are the basic unit of Macrometa streams. Producers publish messages to streams, and consumers process the message and send and acknowledgment. 
```

Comment (diana-macrometa) [reply docs/streams/stream-basics/messages.md]: Remember to spell out on first usage, or just replace the acronym with a word that we do not have to define.

File: docs/streams/stream-basics/messages.md [hunk 13]
```diff
@@ -3,88 +3,34 @@ sidebar_position: 20
...
-
-```js
-stream.consumer("my-subscription", "test.macrometa.io", {
-  subscriptionType: Shared,
-  receiverQueueSize: 1000,
-}
-```
-
-## Message Retention and Expiry
+- **Value -** The data carried by the message. All GDN messages carry raw bytes.
```
Comment (diana-macrometa) [inline docs/streams/stream-basics/messages.md]: This content might be better in a table.

**Review by diana-macrometa — APPROVED** [review]
Looks good. A few more minor suggestions for you to consider, then g2g.

**Pull request merged**